tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -62,13 +65,42 @@ def get_plugin_apps(self):
 
   def is_active(self):
     """The scalars plugin is active iff any run has at least one scalar tag."""
+    if self._db_connection_provider:
+      # The plugin is active if one relevant tag can be found in the database.
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          'SELECT 1 FROM Tags WHERE Tags.plugin_name = ? LIMIT 1',
+          (metadata.PLUGIN_NAME,))
+      return bool(cursor)
+
     if not self._multiplexer:
       return False
 
     return bool(self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME))
 
   def index_impl(self):
     """Return {runName: {tagName: {displayName: ..., description: ...}}}."""
+    if self._db_connection_provider:
+      # Read tags from the database.
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          ('SELECT '
```

Comment:
I would like for us to write SQL queries using multiline strings in go/sqlstyle. For example:

```py
     cursor = db.execute('''
       SELECT
         Tags.tag_name,
         Tags.display_name,
         Runs.run_name
       FROM Tags
       LEFT JOIN Runs
         ON Tags.run_id = Runs.run_id
       WHERE
         Tags.plugin_name = ?
     '''), (metadata.PLUGIN_NAME,))
```

Unfortunately this style guide doesn't seem to be public yet, but we ourselves can still live it. https://github.com/google/styleguide

Reply:
Thanks for formatting it! Done.